### PR TITLE
Feature react 014

### PR DESCRIPTION
**Pull request by pierr**
- Update the component to use `React0.14-rc1`
- Update the `gulpfile.js`

### CODE REVIEW

**Comment by skratchdot** [comment]
Thanks for making these changes!  I just created a new branch (same as you: feature-react-014). I'll pull this into master soon since you didn't update the package.json version yet.  One thing: I've had bugs logged before due to people still wanting to use React v0.12 etc.  My thoughts are to have them use a specific version of this lib in their package.json (so we can stay up-to-date w/ the latest React).  I'd like to hear your thoughts on the best way to version things for people?

Should I pull this in and publish a new version do you think? It'll will probably break some peoples code if I do, and would like to have a response/stance prepared.  I think if they require: `"react-bootstrap-daterangepicker": "0.4.0"` in their package.json then things should continue to work for them.  If they want the new hotness, then they can `"react-bootstrap-daterangepicker": ">0.5.0"

**Comment by pierr** [comment]
As your project is a `React` wrapper around a `jQuery` plugin, which is quite stable , i would say that people who depend on an older version of React shoud use a specific version in their `package.json` file. 
People have to take care of their dependencies.
Maybe, in order to ease transition, you should warn them in the release note.

File: package.json [hunk 1]
```diff
@@ -18,24 +19,25 @@
   "dependencies": {
     "jquery": "1.11.3",
     "moment": ">=2.9.0",
-    "react": ">=0.12 || >=0.13.0-a"
+    "react": "^0.14.0-rc1",
+    "react-bootstrap": "^0.25.100-react-pre.1",
```
Comment (joshhornby) [inline package.json]: is it such a good idea to require a pre release?

**Comment by eskimoblood** [comment]
Any plan when to publish this version on npm?

**Comment by skratchdot** [comment]
@eskimoblood - I was waiting until React was officially released, but I guess I can publish today, and update the release notes.  I'm just a bit worried about breaking existing code, but people can fix that via their package.json files...  As long as the README is updated, I think it should be okay...

**Comment by eskimoblood** [comment]
Maybe you start to follow [Semantic Versioning](http://semver.org/) and increase the major number for every breaking change. Doing this will not create any trouble for someone using your module.

**Comment by skratchdot** [comment]
Well, I guess I am following semantic versioning in that everything in 0.* is considered unstable :)

But I agree.  Maybe I should just start increasing major versions. Problem is, I'm never sure what's going to be big breaking changes until I get a good front-end test suite (maybe start using mocha), so I always find it easier to just stay in 0.* and assume unstable.  That's kinda cheating though I guess!

**Comment by skratchdot** [comment]
After reading the semver FAQ again, and looking at the link you posted, I should just bump the major version today.  I'll add a "rc" string to it since we are still relying on rc versions of React.

**Comment by skratchdot** [comment]
@eskimoblood - I just published `1.0.0-beta.1`.  See `npm view react-bootstrap-daterangepicker`.

**Pull request merged**